### PR TITLE
Fixes #730: document inter-router protocol

### LINKED ISSUE #730: document inter-router protocol

### PR DESCRIPTION
**Pull request by mgoulish**

### CODE REVIEW

File: docs/notes/router-communicates-new-address.adoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+1. When the receiver attaches to router A:
+   qdr_link_inbound_first_attach_CT()
+   calls addr_lookup_handler for an OUTGOING link.
+   (This link is outgoing, because the client is
+    a receiver.)
+
+
+2. The addr_lookup_handler is qcm_addr_lookup_CT().
+   Figure out what the terminus is, based on whether
+   the direction of this link is incoming or ougoing.
```
Comment (ted-ross) [inline docs/notes/router-communicates-new-address.adoc:51]: Typo here.

File: docs/notes/router-communicates-new-address.adoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+=== At Compile Time
+1. The macro QDR_CORE_MODULE_DECLARE declares modstart and defines it to register qcm_mobile_sync_init_CT as the initialization fn for this module.
+
+=== At Startup
+1. The  mobile_sync modstart function is called before main. It calls qdr_register_core_module(), which registers qcm_mobile_sync_init_CT() as the initialization function for this module. 
+
+2. Later during startup, qcm_mobile_sync_init_CT() is called, which subscribes to events of type QDRC_EVENT_ADDR_ADDED_LOCAL_DEST, and registers the callback function qcm_mobile_sync_on_message_CT for messages whose addresses contain 'qdrouter.ma' as their 'qdr field'.
+
+=== At Attach Time
```
Comment (ted-ross) [inline docs/notes/router-communicates-new-address.adoc:114]: It's not clear what is attaching here.  Are you referring to the sender on router B or the receiver on router A?

File: docs/notes/router-communicates-new-address.adoc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+1. The receiver attaches to Router A, which then sends a message to this router (see Router A, above) with the string "qdrouter.ma" in its 'qdr field'.
+
+2. qcm_mobile_sync_on_message_CT is called (see At Startup, above). It detects that this message is a Mobile Address Update, and calls qcm_mobile_sync_on_mau_CT.
+
+3. qcm_mobile_sync_on_mau_CT finds the ID field of the router that this mobile address update came from, and calls qcm_mobile_sync_router_by_id to retrieve that router's node structure. It adds this address to the core's list of addresses that share this treatment (i.e. ANYCAST_CLOSEST, ANYCAST_BALANCED, MULTUICAST_ONCE, etc) and starts up any inbound links associated with this address.
```
Comment (ted-ross) [inline docs/notes/router-communicates-new-address.adoc:120]: Step 3 is glossing over a lot of important detail.  Remote routers are designated by a mask bit.  The address that is created upon receiving the MAU has a bitmask in it which indicates the set of routers on which there is at least one receiver for the address.  If there are already inbound links for the address and this is the first reachable destination, then the inbound links are "started" which means that credit is granted, allowing the senders to begin flowing traffic.

**Review by ted-ross — CHANGES_REQUESTED** [review]
This is a good start in that it shows where to look for various parts of inter-router communication.  I think there are some background topics that might make the sequence easier to express:
 - The data structures that support routing (addresses and nodes)
 - The protocol between the routers (MAR, MAU, sequence numbers, etc.)

**Comment by mgoulish** [comment]
Closing in favor of PR 1387

**Pull request closed without merging**